File: src/main/java/com/teamwizardry/librarianlib/ragdoll/line/PointMass2D.java
package com.teamwizardry.librarianlib.ragdoll.line;

import com.teamwizardry.librarianlib.math.Vec2d;

/**
 * Created by Saad on 12/7/2016.
 */
public class PointMass2D {

    public Vec2d pos, prevPos, nextPos;
    public float mass;
    public boolean pin = false;

    public PointMass2D(Vec2d pos, float mass) {
        this.prevPos = this.pos = pos;
        this.mass = mass;
    }
}

File: src/main/java/com/teamwizardry/librarianlib/ragdoll/line/PointMass2D.kt
package com.teamwizardry.librarianlib.ragdoll.line

import com.teamwizardry.librarianlib.math.Vec2d

/**
 * Created by Saad on 12/7/2016.
 */
class PointMass2D(pos: Vec2d, var mass: Float) {

    var pos: Vec2d = pos
    var prevPos: Vec2d = pos
    var nextPos: Vec2d? = null
    var pin = false
}
